modules/auxiliary/dos/windows/http/rb_dos_iis_2021_31166.rb

```diff
@@ -0,0 +1,117 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Dos
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows IIS HTTP Protocol Stack DOS',
+        'Description' => %q{
+          This module is a proof of concept for CVE-2021-31166 ("HTTP Protocol
+          Stack Remote Code Execution Vulnerability") to exploit a DOS attack,
+          a use-after-free dereference in http.sys patched by Microsoft
+          in May 2021. 

```

Comment:
```suggestion
          This module exploits CVE-2021-31166, a UAF bug in http.sys
          that was patched by Microsoft in May 2021, to cause a BSOD and
          crash the target IIS server. Successful exploitation will result in the target
          BSOD'ing and the target server rebooting.
```

Reply:
Suggested rewording. I don't think its appropriate to say its a stack vulnerability when its a heap vulnerability as use-after-free is a heap based vulnerability, not a stack based vulnerability. You free allocations from the heap, not the stack (there are some weird cases when one uses the stack as heap memory but I don't think they are applicable here).